tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -787,31 +805,57 @@ module rec Compiler =
 
     let compileExeAndRun = asExe >> compileAndRun
 
-    let private evalFSharp (fs: FSharpCompilationSource) : CompilationResult =
-        let source = fs.Source.GetSourceText |> Option.defaultValue ""
-        let options = fs.Options |> Array.ofList
-
-        use script = new FSharpScript(additionalArgs=options)
-        let (evalResult: Result<FsiValue option, exn>), (err: FSharpDiagnostic[]) = script.Eval(source)
-        let diagnostics = err |> fromFSharpDiagnostic
+    let private processScriptResults fs (evalResult: Result<FsiValue option, exn>, err: FSharpDiagnostic[])  =
+        let perFileDiagnostics = err |> fromFSharpDiagnostic
+        let diagnostics = perFileDiagnostics |> List.map snd
+        let (errors, warnings) = partitionErrors diagnostics
         let result =
             { OutputPath   = None
               Dependencies = []
               Adjust       = 0
-              Diagnostics  = diagnostics
+              Diagnostics  = if fs.IgnoreWarnings then errors else diagnostics
```

Comment:
TBH the whole naming becomes a bit misleading here, diagnostics are errors and errors are diagnostics. I'd just propagate the existing naming and build on top of it even if it is going to be cumbersome.

Reply:
This PR does not change it.
Diagnostics are the resulting diagnostics tailored to the configured settings. So if someone decides to ignore warning, they are excluded from it.

Reply:
Well before `Diagnostics  = diagnostics`, now `PerFileErrors = diagnostics` and `Diagnostics   = diagnostics |> List.map snd`. That's what I meant. Not the end of the world